engine/access/rest/handler.go

```diff
@@ -36,19 +37,22 @@ type Handler struct {
 	backend        access.API
 	linkGenerator  models.LinkGenerator
 	apiHandlerFunc ApiHandlerFunc
+	chain          flow.Chain
```

Comment:
The chain ID is needed when converting the incoming JSON transaction to a flow.Transaction. It is used to validate if the address belongs to the chain. This is similar to the gRPC implementation which also needs the chain id to be passed in.